### PR TITLE
[WPB-5603] Fix the team member deleted event reason

### PR DESCRIPTION
**Pull request by mdimjasevic**
As reported by a client developer, the event reason when a team member is deleted was incorrectly "member removed", when it should have been "member deleted" (from a team). This PR fixes it.

Tracked by https://wearezeta.atlassian.net/browse/WPB-5603.

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: libs/wire-api/test/golden/Test/Wire/API/Golden/Generated/Event_conversation.hs [hunk 1]
```diff
@@ -35,6 +35,7 @@ import Wire.API.Conversation.Code
 import Wire.API.Conversation.Role
 import Wire.API.Conversation.Typing
 import Wire.API.Event.Conversation
+import Wire.API.Event.LeaveReason
```
Comment (fisx) [inline libs/wire-api/test/golden/Test/Wire/API/Golden/Generated/Event_conversation.hs:38]: also add a manual golden test for `ConversationRemoveMembers`?

File: libs/wire-api/src/Wire/API/Event/LeaveReason.hs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import Data.Schema
+import Imports
+import Wire.Arbitrary
+
+--  | The reason for a member to leave
+--    There are three reasons
+--    - the member has left on their own
+--    - the member was removed from the team
+--    - the member was removed by another member
+data EdMemberLeftReason
```
Comment (fisx) [inline libs/wire-api/src/Wire/API/Event/LeaveReason.hs:29]: Moving and renaming of this data type could have gone to a separate PR (but I'm very bad at following through with this idea myself :).

File: libs/wire-api/src/Wire/API/Conversation/Action.hs [hunk 3]
```diff
@@ -130,11 +130,7 @@ conversationActionSchema SConversationLeaveTag =
     "ConversationLeave"
     (S.description ?~ "The action of some users leaving a conversation on their own")
     $ pure ()
-conversationActionSchema SConversationRemoveMembersTag =
-  objectWithDocModifier
-    "ConversationRemoveMembers"
-    (S.description ?~ "The action of some users being removed from a conversation")
-    $ field "targets" (nonEmptyArray schema)
+conversationActionSchema SConversationRemoveMembersTag = schema
```
Comment (fisx) [inline libs/wire-api/src/Wire/API/Conversation/Action.hs:133]: nice :)

**Review by fisx — APPROVED** [review]
LGTM!  I think concourse is having trouble keeping the federation network running and connected, good luck with that...  :/

**Review by pcapriotti — APPROVED** [review]

File: libs/wire-api/test/golden/Test/Wire/API/Golden/Generated/Event_conversation.hs (hunk 1 above)
Comment (mdimjasevic) [reply libs/wire-api/test/golden/Test/Wire/API/Golden/Generated/Event_conversation.hs:38]: Thanks for spotting this! I added them in 26b485c7f.

File: libs/wire-api/src/Wire/API/Event/LeaveReason.hs (hunk 2 above)
Comment (mdimjasevic) [reply libs/wire-api/src/Wire/API/Event/LeaveReason.hs:29]: True. It covers the majority of the files changed in this PR, only for importing the module. Still, I made it as part of this PR given that it's a simple change and a tiny new module.

**Pull request merged**